task.php

```diff
@@ -0,0 +1,21 @@
+<?php
+header('Content-Type: application/json; charset=utf-8');
+header("Access-Control-Allow-Origin: *");
+require 'vendor/autoload.php';
+
+try {
+    $db = \ProjectManager\Services\DbConnector::connect();
+    if (isset($_GET['id']) && ctype_digit($_GET['id'])) {
+        try {
+            $data = \ProjectManager\Hydrators\TasksHydrator::getTaskByUserAndProjectId($db, $_GET['id']);
+            $jsonData =  \ProjectManager\Services\ConvertToJsonService::convert($data, \ProjectManager\Services\ConvertToJsonService::PROJECT_SUCCESS_MESSAGE);
+        } catch (Throwable $e) {
+            $jsonData = \ProjectManager\Services\ConvertToJsonService::invalidTaskIdResponse();
+        }
```

Comment:
You shouldnt need this catch. this is catching the `TypeError` from `ConvertToJsonService::convert` when you pass `$data` as false because there is no Task. instead you should use an if statement to check it isnt empty, in the else, you can do the invalid task response.